### PR TITLE
Fix 47 unresolved rustdoc link warnings across 14 files (closes #19)

### LINKED ISSUE #19: Documentation CI: unresolved rustdoc links in inference/snake.rs and multi_agent/{messages,joint,mod}.rs
## Symptom

The `Documentation` job in CI fails with rustdoc "unresolved link" errors in four files:

- `src/inference/snake.rs`
- `src/multi_agent/messages.rs`
- `src/multi_agent/joint.rs`
- `src/multi_agent/mod.rs`

Likely caused by rustdoc bracket-link syntax (`[Foo]`) referencing types that have been renamed, moved, or aren't in scope from where the doc comment lives.

## Reproduction

```bash
cargo +nightly doc --features training --no-deps --document-private-items 2>&1 | grep -E 'unresolved|warning'
```

## Acceptance

- `cargo +nightly doc --features training --no-deps` produces no `unresolved link` warnings (or, if the project policy is to treat warnings as errors in CI, the Documentation CI job goes green).
- All bracketed type/function references in the four files resolve to real items, either via fully-qualified paths or by adding `use` imports in the appropriate scope (or by removing the bracket if the reference was unintentional).

## Discovery

Flagged by the Loom Judge during re-review of PR #16 ([comment](https://github.com/rjwalters/thrust/pull/16#issuecomment-4455152478)).

### PR DESCRIPTION
**Pull request by rjwalters**
## Summary

Closes #19.

Fixes all 47 `unresolved link` rustdoc warnings reported by the curator's audit on `cargo +nightly doc --no-deps --all-features`. Since CI runs with `RUSTDOCFLAGS="-D warnings"`, every one of these was a hard failure for the Documentation job.

### Scope note: expanded from 4 files to 14 files

The original issue title listed only 4 files. The curator audit revealed the same root cause across 14 files (47 warnings total). All 47 warnings have to land together — fixing only the originally-cited subset would still leave CI red.

### Changes

**Group A — shape-annotation false positives (42 warnings).** Doc comments like ``"Action logits [num_actions]"`` were being parsed by rustdoc as intra-doc links to an item named `num_actions`. Wrapped each bracketed shape annotation in inline-code backticks so rustdoc treats them as code rather than links. Affects:

- `src/buffer/rollout.rs`, `src/buffer/rollout/{gae,sampling,storage}.rs`
- `src/inference/snake.rs`
- `src/multi_agent/messages.rs`
- `src/policy/{inference,universal_inference}.rs`
- `src/train/ppo/{loss,trainer}.rs`
- `src/utils/normalize.rs`

**Group B — genuine broken references (5 warnings).**

- `src/multi_agent/mod.rs` (3 warnings): `[`PolicyLearner`]`, `[`Population`]`, `[`GameSimulator`]` in the `//!` module doc resolved against `multi_agent`'s scope, where these names only exist inside submodules. Replaced with fully-qualified `[`crate::multi_agent::learner::PolicyLearner`]` etc. Also rewrote `[`joint::JointMultiAgentTrainer`]` and `[`joint`]` (declared *after* the `//!` block) as `[`crate::multi_agent::joint::JointMultiAgentTrainer`]` / `[`crate::multi_agent::joint`]`.
- `src/policy/multi_discrete_mlp.rs`: replaced unqualified `[`MlpPolicy`]` with `[`crate::policy::mlp::MlpPolicy`]`.
- `src/multi_agent/joint.rs:339`: **noteworthy finding** — the doc comment referenced `[`Self::with_adam`]`, but `with_adam` does not exist on `JointMultiAgentTrainer` (only `new`, `device`, `collect_rollout`, `update`). This is a stale reference, most likely left over from PR #15. Per the curator's recommendation, deleted the sentence promising the helper rather than synthesising a method just to make the link resolve. If `with_adam` is genuinely planned, a follow-up issue is the right tracker for it.

### Verification

```
$ cargo +nightly doc --features training --no-deps 2>&1 | grep -c "unresolved link"
0
$ cargo +nightly doc --all-features --no-deps 2>&1 | grep -c "unresolved link"
0
```

Total rustdoc warning count drops from 194 to 147; the remaining 147 are pre-existing `missing documentation for ...` warnings on `pub` items (e.g. `src/optimize/space.rs:38,40`) that predate this branch and are **out of scope** for #19. They are a separate hygiene issue — the Documentation job will still fail until they're cleaned up — but the unresolved-link class of failure is now eliminated.

## Test plan

- [x] `cargo +nightly doc --features training --no-deps` shows zero `unresolved link` warnings
- [x] `cargo +nightly doc --all-features --no-deps` shows zero `unresolved link` warnings
- [x] `cargo +nightly check --all-features` still compiles (no syntactic damage from edits)
- [x] `cargo +nightly fmt` applied
- [ ] CI `Documentation` job: will still fail because of the pre-existing `missing_docs` warnings, but the unresolved-link warnings should be gone from the log

### CODE REVIEW

**Comment by rjwalters** [comment]
## LGTM — Approved

Verified all acceptance criteria locally on the worktree (`LIBTORCH=/Users/rwalters/GitHub/thrust/libtorch`):

- `cargo +nightly doc --features training --no-deps 2>&1 | grep -c "unresolved link"` → **0**
- `cargo +nightly doc --all-features --no-deps 2>&1 | grep -c "unresolved link"` → **0**
- `cargo +nightly check --all-features` → compiles cleanly

### Spot checks

**Group A (42 shape-annotation false positives):** Sampled `src/buffer/rollout/storage.rs`, `src/buffer/rollout/sampling.rs`, `src/train/ppo/{loss,trainer}.rs`, `src/utils/normalize.rs`, `src/policy/inference.rs`. All bracketed shape annotations are now wrapped in inline-code backticks (e.g. `` `[batch_size]` ``, `` `[num_envs]` ``, `` `[batch_size][obs_dim]` ``). Pattern is consistent across files.

**Group B (5 genuine references):**

1. `src/multi_agent/mod.rs` — rewrote `PolicyLearner`, `Population`, `GameSimulator`, `joint::JointMultiAgentTrainer`, `joint` to fully-qualified `crate::multi_agent::*` paths. Verified each target type exists at the claimed submodule path:
   - `learner::PolicyLearner` → `src/multi_agent/learner.rs:24`
   - `population::Population` → `src/multi_agent/population.rs:16`
   - `simulator::GameSimulator` → `src/multi_agent/simulator.rs:25`
   - `joint::JointMultiAgentTrainer` → `src/multi_agent/joint.rs:319`

2. `src/policy/multi_discrete_mlp.rs` — `MlpPolicy` rewritten to `crate::policy::mlp::MlpPolicy`. Verified: `pub struct MlpPolicy` is declared in `src/policy/mlp.rs:71`, and `src/policy/mod.rs:10` has `pub mod mlp;`. Path is correct.

3. `src/multi_agent/joint.rs:335-340` — confirmed `Self::with_adam` reference was removed (not fabricated). The surrounding doc text now reads coherently:
   > `optimizers` must have one entry per policy and be configured for the caller's choice of learning rate / weight decay. All policies must be on the same device.
   No dangling phrasing or orphaned sentence fragments. Good call on the curator's recommendation — synthesizing a method just to make rustdoc happy would have been worse.

### CI status (mergeStateStatus: UNSTABLE)

CI is red, but **the failures are all pre-existing on `main`** (`2027f61`):

- **Documentation** — fails on `missing documentation for ...` errors only; **zero `unresolved link` errors** remain (verified by grepping the job log). This is exactly the out-of-scope class called out in the PR description and review brief.
- **Tests (Linux + libtorch)** & **Tests (macOS + libtorch)** — fail compiling `examples/games/cartpole/train_cartpole_modern.rs` and `train_cartpole_long.rs` due to stale `RolloutBatch.log_probs`/`.values` field references, a missing `hyperparameters` field on `TrainingMetadata`, and a `Vec<f32>` vs `&[f32]` arg mismatch on `RolloutBuffer::add`. This PR doesn't touch `examples/` at all (`git diff main..HEAD -- examples/` returns nothing), and `main` HEAD has the identical failure profile on these jobs. Worth a separate issue.
- **Security Audit** — also failing on main HEAD.

The Documentation job log no longer contains a single `unresolved link` mention, which is the entire scope of issue #19. Approving on that basis.

Nice work on the curator's scope expansion + the principled refusal to fabricate `with_adam`. Both calls were correct.
